extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ExternalColumnSelectorFactory.java

```diff
@@ -97,7 +103,11 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
       public Object getObject()
       {
         try {
-          return delegateDimensionSelector.getObject();
+          ExpressionType expressionType = ExpressionType.fromColumnType(dimensionSpec.getOutputType());
```

Comment:
should probably save this outside of getObject so don't have to recreate every time (it uses interned objects so it won't be making new ones really, but could still save some overhead)